### ISSUE TITLE
import_css blocks the main thread

### ISSUE THREAD

**Issue by memecode** [litehtml/litehtml#296]
To implement the litehtml::document_container::import_css function:

    void import_css(litehtml::string& text, const litehtml::string& url, litehtml::string& baseurl);

The application has to go out to the network and get the CSS and then at some later time, fill out 'text' with the content. That doesn't work if the network is slow to respond, taking seconds to complete... 

A better way of doing it would be to pass a callback for the first parmeter... like:

    void import_css(std::function<void(litehtml::string& text)> callback, const litehtml::string& url, litehtml::string& baseurl);

Then the implementor can go out setup a thread... pass the callback... do the network stuff... get the file... THEN call the callback. Maybe with some extra glue to make it run in the same thread as the caller.

litehtml just waits for the callback and applies the changes. No waiting around.

**Comment by tordex** [comment]
I agree with you. Right now I'm rewriting [litebrowser-linux](https://github.com/litehtml/litebrowser-linux) with multithreading support and found the css loading blocks. I'll try to solve this issue.
